src/backend/distributed/utils/metadata_cache.c

```diff
@@ -677,6 +722,32 @@ InvalidateDistRelationCacheCallback(Datum argument, Oid relationId)
 			cacheEntry->isValid = false;
 		}
 	}
+
+	/* now check whether we're in the middle of dropping the extension */
+
+	/* if we're not in a transaction then we'll crash while looking up oids */
+	inTransaction = IsTransactionState();
+	if (!inTransaction)
+	{
+		return;
+	}
+
+	shardPlacementId = DistShardPlacementRelationId();
+
+	elog(WARNING,"Invalidate rel: %d target: %d", relationId, shardPlacementId);
+
+	if (relationId != InvalidOid && relationId == shardPlacementId)
+	{
+		shardPlacementId = get_relname_relid("pg_dist_shard_placement",
+											 PG_CATALOG_NAMESPACE);
+		if (shardPlacementId == InvalidOid)
+		{
+			/* The extension is being dropped, we should drop our state too */
+			elog(WARNING,"Dropping all metadata_cache state");
+
+			memset(&metadataCacheState, 0, sizeof(MetadataCacheState));
```

Comment:
It's true. I wouldn't ever use this line as an argument to avoid adding a field to the struct which doesn't have an initial value represented as 0, but for now it's pretty convenient and I think simpler.

Actually, this means grepping for `extensionLoaded`, for instance, will miss this access. That's probably reason enough to be more explicit.